src/Widget/Attribute/FieldAttributes.php

```diff
@@ -18,53 +17,123 @@
 trait FieldAttributes
 {
     private bool $ariaDescribedBy = false;
-    private string $attribute = '';
-    private array $attributes = [];
     private string $containerClass = '';
+    /** @psalm-var array<string, string> */
+    private array $containersClass = [];
     private string $errorClass = '';
+    /** @psalm-var array<string, string> */
+    private array $errorsClass = [];
     private string $errorMessage = '';
     private string $hintClass = '';
+    /** @psalm-var array<string, string> */
+    private array $hintsClass = [];
     private string $id = '';
     private string $inputClass = '';
+    private array $inputsClass = [];
     private string $labelClass = '';
+    /** @psalm-var array<string, string> */
+    private array $labelsClass = [];
     private string $invalidClass = '';
+    /** @psalm-var array<string, string> */
+    private array $invalidsClass = [];
     private string $validClass = '';
+    /** @psalm-var array<string, string> */
+    private array $validsClass = [];
     private array $parts = [];
     private string $template = "{label}\n{input}\n{hint}\n{error}";
+    /** @psalm-var array<string, string> */
+    private array $templates = [];
+    private string $type = '';
     private string $validationStateOn = 'input';
     private ?FormModelInterface $formModel = null;
 
     /**
-     * Set aria-describedby attribute.
+     * Set invalid class each for field type.
+     *
+     * @param array $invalidClass the input class to be used to layout the field.
+     *
+     * ```php
+     * [Field::TYPE_TEXT => 'test-class-1', Field::TYPE_SUBMIT_BUTTON => 'test-class-2']
      *
      * @return static
      *
-     * @link https://www.w3.org/TR/WCAG20-TECHS/ARIA1.html
+     * @psalm-param array<string, string> $invalidClass
      */
-    public function ariaDescribedBy(): self
+    public function addInvalidClass(array $invalidClass): self
```

Comment:
```suggestion
    public function invalidClass(array $invalidClass): self
```

Since we're not adding a class. We are setting and it for a cloned instance. Same applies to other similar `add*` methods.

Reply:
fixed.